source/OptimaJet.DWKit.StarterApplication/Services/BuildEngine/BuildEngineBuildService.cs

```diff
@@ -231,36 +231,47 @@ protected BuildResponse GetBuildEngineBuild(Product product)
         protected async Task BuildCompletedAsync(Product product, BuildResponse buildEngineBuild)
         {
             Log.Information($"BuildCompletedAsync: product={product.Id}, response:Id={buildEngineBuild.Id},Status={buildEngineBuild.Status},Result={buildEngineBuild.Result},Date={buildEngineBuild.Updated}");
-            DateTime? mostRecentArtifactDate = new DateTime(2018, 10, 1);
             ClearRecurringJob(product.Id);
-            if (buildEngineBuild.Artifacts != null)
+            await SaveArtifacts(product, buildEngineBuild);
+            var messageParms = new Dictionary<string, object>()
             {
-                var productBuild = await ProductBuildRepository.Get()
-                    .Where(pb => pb.ProductId == product.Id && pb.BuildId == product.WorkflowBuildId)
-                    .FirstOrDefaultAsync();
-                foreach(KeyValuePair<string, string> entry in buildEngineBuild.Artifacts)
+                { "projectName", product.Project.Name },
+                { "productName", product.ProductDefinition.Name}
+            };
+            await SendNotificationSvc.SendNotificationToUserAsync(product.Project.Owner, "buildCompletedSuccessfully", messageParms);
+            await UpdateProductBuild(buildEngineBuild, product, true);
+        }
+
+        private async Task SaveArtifacts(Product product, BuildResponse buildEngineBuild)
+        {
+            DateTime? mostRecentArtifactDate = new DateTime(2018, 10, 1);
```

Comment:
2018/10/1?

You can do `= new DateTime()` which assigns the DateTime with the DateTime.MinValue

https://docs.microsoft.com/en-us/dotnet/api/system.datetime.minvalue?view=netcore-2.1

Reply:
Just moved the existing code from old line 234.  I expected this was probably because it was about the time the project started and someone picked that date for that reason.